File: server/src/main/java/com/pn/oupia/model/Asset.java
package com.pn.oupia.model;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@Table(name = "asset")
@NoArgsConstructor
public class Asset implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "asset_name", length = 255)
    private String assetName;

    @Column(name = "asset_slug", unique = true, length = 300)
    private String assetSlug;

    @Column(name = "asset_description", length = 2000)
    private String assetDescription;

    @Column(name = "full_location", length = 300)
    private String fullLocation;

    @Column(name = "location_lat")
    private Float locationLat;

    @Column(name = "location_long")
    private Float locationLong;

    @Column(name = "is_deleted")
    private Boolean isDeleted;

    @Column(name = "created_at", insertable = true, updatable = false)
    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;

    @Column(name = "updated_at", insertable = false, updatable = true)
    @Temporal(TemporalType.TIMESTAMP)
    private Date updatedAt;

    @JoinColumn(name = "asset_type_id", referencedColumnName = "id")
    @ManyToOne(cascade = CascadeType.PERSIST)
    private AssetType assetType;

    @JoinColumn(name = "user_id", referencedColumnName = "id")
    @ManyToOne(cascade = CascadeType.PERSIST)
    private User user;

    @JsonIgnore
    @OneToMany(cascade = { CascadeType.PERSIST, CascadeType.REMOVE }, mappedBy = "asset")
    private List<Unit> units;

    @PrePersist
    protected void onCreate() {
        createdAt = new Date();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = new Date();
    }
}

File: lib/interfaces/response/Asset.ts
import { AssetType } from "@/lib/enums";
import { Amenity } from "@/lib/types";

export interface IAssetItem {
    id: number,
    assetName: string,
    assetSlug: string,
    assetDescription: string,
    fullLocation: string,
    locationLat: number,
    locationLong: number,
    assetType: string,
    userId: number,
    price: number,
    area: number,
    maxPeople: number,
    amenities: Amenity[],
    images: string[],
    isDeleted: boolean,
    createdAt: Date,
    updatedAt: Date
}

export interface IAssetInformation {
    assetName: string,
    assetDescription: string,
    assetType: AssetType,
    fullLocation: string,
    locationLat: number,
    locationLong: number,
    price: string,
    area: string,
    maxPeople: string,
    amenities: Amenity[],
    images: File[],
    businessLicense: File
}